src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,55 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.addAlive(this);
+  }
+
+  static addAlive(animal) {
+    Animal.alive.push(animal);
+  }
+
+  static getAlive() {
+    return Animal.alive;
+  }
+
+  static removeAlive(animal) {
+    Animal.alive = Animal.alive.filter((item) => item !== animal);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name);
+    this.health = health;
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name);
+    this.health = health;
+  }
+
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
+      Animal.removeAlive(animal);
+    }
```

Comment:
[CODE PERFORMANCE] - It is preferred to filter animals with health less than or equal to 0 from `Animal.alive` array using Array's `filter` method. Instead of checking the animal's health after each bite, you could have a separate method that cleans up dead animals from the `Animal.alive` array. This would avoid the need to check the health of each animal after every bite, which could improve performance if there are many bite operations occurring.